framework/decode/vulkan_virtual_swapchain.cpp

```diff
@@ -130,103 +163,260 @@ VkResult VulkanVirtualSwapchain::GetSwapchainImagesKHR(PFN_vkGetSwapchainImagesK
         swapchain          = swapchain_info->handle;
         replay_image_count = &swapchain_info->replay_image_count;
 
+        // Get the private data so we have access to the virtual swapchain-specific information.
+        if (private_data_.find(swapchain) == private_data_.end())
+        {
+            GFXRECON_LOG_ERROR(
+                "Virtual swapchain vkGetSwapchainImagesKHR missing private data for swapchain (ID = %" PRIu64 ")",
+                swapchain_info->capture_id);
+        }
+        private_data = private_data_[swapchain];
+        assert(private_data != nullptr);
+
         if (images != nullptr)
         {
-            replay_swapchain_images.resize(*replay_image_count);
+            private_data->replay_swapchain_images.resize(*replay_image_count);
         }
     }
 
     // TODO: Adjust the swapchain image format if the specified format is not supported by the replay device.
 
-    auto result = func(device, swapchain, replay_image_count, replay_swapchain_images.data());
+    result = func(device, swapchain, replay_image_count, private_data->replay_swapchain_images.data());
+
+    if (static_cast<uint32_t>(private_data->capture_to_replay_index.size()) != capture_image_count)
```

Comment:
I think in general the `static_cast` here should be on `capture_image_count` instead, up to `size_t`, because the result of `size` may be 64 bits and so the static cast on `size()` loses information.  It seems so unlikely though, that size() would return > 2^32 that maybe I don't care if you don't have time to make the change.

Reply:
Done